### DISCUSSION TITLE
Org-cite stuck in minibuffer

### DISCUSSION THREAD

**Discussion by mok0** [minad/corfu#460] [Q&A] [answered]
When inserting a citation using the function `org-cite-insert ` or using the key binding `C-c C-x @`, the minibuffer will list the entries in the `.bib` file, but it is not possible to select a reference like you normally can with <RET>. Hitting <RET> just keeps adding references. 

By Googling I can see that in other completion frameworks there is a key combo to terminate org-cite and insert the reference(s): `C-<RET>` for Helm and `C-M-j` for Ivy, but I can't find a solution for corfu.

In any case I think it's a bad design that org-cite doesn't allow standard completion in Emacs. When I type the `[cite:@blabla]` stanza manually in Org-mode it works, but that kinda defeats the purpose of having an interface.

**Comment by mok0** [comment, marked as answer]
**I found a solution** myself, but I don't understand why it works. It seems after invoking `org-cite-insert`, I can insert the references I want, and end the selection by typing `M-<RET>`.  When I ask Emacs what this key-binding corresponds to, it tells me `org-insert` and normally in Org that is used to insert a new headline. 

I think it would make more sense if `org-cite` didn't hog `<RET>` but let Corfu keep it to end the selection, and instead used another key combo, e.g. `M-<RET>` if you wanted to add more references. In other words, swap the two key-bindings, unfortunately I don't know how to do that.

**Reply by minad** [reply]
Note that you are talking about **Vertico** here, not about Corfu.

> I think it would make more sense if org-cite didn't hog <RET> but let Corfu keep it to end the selection, and instead used another key combo, e.g. M-<RET> if you wanted to add more references. In other words, swap the two key-bindings, unfortunately I don't know how to do that.

Yes, this would be better. This is indeed possible by passing the empty string as default argument to `completing-read`. But then, if you press RET without having inserted anything yet, it would just quit right away. As a more sophisticated solution, one could use RET first to insert the first reference, and for the following steps, reverse the meaning.
